Checkbox is no longer broken

SebVortex opened the pull request:
It can be checked and unchecked now.

## Description
Unable to select the checkbox "I'll be taking classes outside of the U.S. and U.S. territories" for the radio button "Other location" inside the accordion "Learning format and location"

## Original issue(s)
[department-of-veterans-affairs/va.gov-team#33608](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/33608)

## Testing done
Confirmation that the checkbox checks & unchecks when clicked.

## Screenshots
Unchecked
<img width="436" alt="Unchecked" src="https://user-images.githubusercontent.com/86262790/146985215-d69ded4e-41ae-40fa-9104-c2f4a355de2d.png">

Checked
<img width="434" alt="Checked" src="https://user-images.githubusercontent.com/86262790/146985211-02ffcba8-31ee-4ff9-8350-430cb2a2849d.png">


## Acceptance criteria
- [x] The checkbox indicating foreign classes will be taken functions when clicked

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

SebVortex committed:
Checkbox is no longer broken

It can be checked and unchecked now.

jerekshoe commented on src/applications/gi-sandbox/components/profile/CalculateYourBenefitsForm.jsx:
```diff
@@ -842,7 +843,7 @@ function CalculateYourBenefitsForm({
           onChange={handleHasClassesOutsideUSChange}
-          checked={inputs.classesOutsideUS}
+          checked={classesOutsideUSChecked}
           name={'classesOutsideUS'}
```
Sorry to butt in but the reason the checkbox doesn't check is because the `Checkbox` component turns this name prop into `classesoutsideus`, so the `field` value (e.target.name) when calling `calculatorInputChange` above is actually `classesoutsideus`, not `classesOutsideUS` like you would expect. Because of this `inputs.classesOutsideUS` is always false, and the checkbox remains unchecked

SebVortex replied:
Ah. Interesting quirk. I'll make the change so an extra property won't be required.

jerekshoe replied:
You will want to update your reducer (reducers/calculator.js) if you want to change the property. Another option is to update the `handleHasClassesOutsideUSChange` to call `handleCheckboxChange('classesOutsideUS', e.target.value)` and update `handleCheckboxChange` to accept a field and value as parameters instead of an event.

SebVortex replied:
I just changed the checked property to inputs.classesoutsideus & the earlier reference to control zip code input presentation to the same lowercased property and it seems to work correctly.

Updated screenshots show it affects presentation as well too (neglected this before as it wasn't in the description of expected behavior but this behavior looks correct).

jerekshoe replied:
You might want to make sure that the selectors (selectors/calculator.js) still work as well

SebVortex committed:
Checkbox is no longer broken

Turns out checkbox lowercases name for the target name. Camelcased names will always be false in checkboxes.

SebVortex committed:
Selector/calculator.js uses the same input prop

Changed "outsideUS" property used in the checkbox here as well to ensure that the calculator is using the same input property.

Everything appears to be working as expected.
